File: selection sort.cpp
/*******************************************************************************
*
* Program: Selection Sort
*
*******************************************************************************/

#include <iostream>

using namespace std;

void selectionSort(int arr[], int length);

int main()
{
  // A test array and array length
  int arr[] = {6, 3, 2, 4, 1, 5};
  int length = 6;
  
  // Call the function to perform the selection sort algorithm
  selectionSort(arr, length);
  
  // Print out the array so we can be sure it is sorted correctly
  for (int i = 0; i < length; i++)
  {
    cout << "arr[" << i << "] = " << arr[i] << endl;
  }
  return 0;
}

void selectionSort(int arr[], int length)
{
    int i, j, min;
    for (i = 0; i < length - 1; i++) {
        // Find the minimum element in unsorted array
        min = i;
        for (j = i + 1; j < length; j++) {

            if (arr[j] < arr[min])
                min = j;
        }
 
        // Swap the found minimum element
        // with the first element

        if (min != i)
        {
            //swap(arr[min], arr[i]);
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
    }
}

File: Selection Sort.py
# -*- coding: utf-8 -*-
"""
Created on Sun Dec 24 12:17:56 2023

@author: youreternity1997
"""

# Selection sort in Python

def selectionSort(array, size):
   
    for i in range(size):
        min_idx = i

        for j in range(i + 1, size):
            
            # select the minimum element in each loop
            if array[j] < array[min_idx]:
                min_idx = j
         
        # put min at the correct position
        (array[i], array[min_idx]) = (array[min_idx], array[i])


data = [-22, 13,-1, 42, 45, 0, 11, -9, 33]
size = len(data)
selectionSort(data, size)
print('Sorted Array in Ascending Order:')
print(data)